src/main/java/sirius/biz/tycho/kb/SynchronizeArticlesTask.java

```diff
@@ -0,0 +1,248 @@
+import sirius.pasta.tagliatelle.Template;
+import sirius.pasta.tagliatelle.rendering.GlobalRenderContext;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+
+/**
+ * Scans all available articles and turns them into {@link KnowledgeBaseEntry knowledge base entries}.
+ * <p>
+ * Note that this also deletes outdated entries or empty chapters.
+ */
+@Register(framework = KnowledgeBase.FRAMEWORK_KNOWLEDGE_BASE)
+public class SynchronizeArticlesTask implements EndOfDayTask {
+
+    private boolean executed = false;
+
+    @Part
+    private Locks locks;
+
+    @Part
+    private KeyGenerator keyGenerator;
+
+    @Part
+    private Elastic elastic;
+
+    @Part
+    private Mixing mixing;
+
+    @Part
+    private Tagliatelle tagliatelle;
+
+    @Part
+    private KnowledgeBase knowledgeBase;
+
+    @Override
+    public String getName() {
+        return "synchronize-knowledgebase";
+    }
+
+    @Override
+    public void execute() throws Exception {
+        if (executed) {
+            return;
+        }
+
+        synchronizeArticles();
+
+        if (!Sirius.isDev()) {
+            executed = true;
+        }
+    }
+
+    private void synchronizeArticles() {
+        String syncId = keyGenerator.generateId();
+        Sirius.getClasspath()
+              .find(Pattern.compile("(default/|customizations/[^/]+/)?kb/.*\\.pasta"))
+              .forEach(matcher -> updateArticle(cleanupTemplatePath(matcher.group(0)), syncId));
+
+        elastic.getLowLevelClient().refresh(mixing.getDescriptor(KnowledgeBaseEntry.class).getRelationName());
+        cleanupOldEntries(syncId);
+        if (!Sirius.isDev()) {
+            // We only remove empty chapters from production instances as in development systems it might
+            // be helpful to see these chapters to know their ID.
+            cleanupEmptyChapters();
+        }
+        checkCrossReferences();
+        knowledgeBase.resetLanguages();
+    }
+
+    private String cleanupTemplatePath(String templatePath) {
+        if (templatePath.startsWith("default/")) {
+            return "/" + templatePath.substring(8);
+        }
+        return "/" + templatePath;
+    }
+
+    private void updateArticle(String templatePath, String syncId) {
+        try {
+            Template template = tagliatelle.resolve(templatePath).orElseThrow(() -> new IllegalArgumentException(""));
+            GlobalRenderContext context = tagliatelle.createRenderContext();
+            template.render(context);
+
+            String articleId = Value.of(context.getExtraBlock("code")).toUpperCase();
+            String lang = context.getExtraBlock("lang");
+            KnowledgeBaseEntry entry = findOrCreateEntry(templatePath, articleId, lang);
+
+            entry.setSyncId(syncId);
+            entry.setChapter(Value.of(context.getExtraBlock("chapter")).asBoolean());
```

Comment:
Frage: Könnte man hier nicht die Mappings aus dem Entity nehmen um robuster die Sachen ausm JSON zu holen?

Also zB: `entry.setChapter(Value.of(context.getExtraBlock(KnowledgeBaseEntry.CHAPTER.getName())).asBoolean());`